src/Features/Core/Portable/CodeRefactorings/AbstractRefactoringHelpersService.cs

```diff
@@ -300,52 +345,111 @@ private static TextSpan GetSpanWithoutAttributes(SyntaxNode node, SyntaxNode roo
         /// </para>
         /// </summary>
-        protected virtual SyntaxNode DefaultNodeExtractor(SyntaxNode node, ISyntaxFactsService syntaxFacts, bool extractParentsOfHeader)
+        protected virtual IEnumerable<SyntaxNode> DefaultNodesExtractor(SyntaxNode node, ISyntaxFactsService syntaxFacts, bool extractParentsOfHeader)
         {
             // REMARKS: 
             // The set of currently attempted extractions is in no way exhaustive and covers only cases
             // that were found to be relevant for refactorings that were moved to `TryGetSelectedNodeAsync`.
             // Feel free to extend it / refine current heuristics. 
 
+            foreach (var extractedNode in ExtractNodesSimple(node, syntaxFacts))
+            {
+                yield return extractedNode;
+            }
+
+            if (extractParentsOfHeader)
+            {
+                foreach (var headerNode in ExtractNodesOfHeader(node, syntaxFacts))
+                {
+                    yield return headerNode;
+                    foreach (var extractedNode in ExtractNodesSimple(headerNode, syntaxFacts))
+                    {
+                        yield return extractedNode;
+                    }
+                }
+            }
+        }
+
+        protected virtual IEnumerable<SyntaxNode> ExtractNodesSimple(SyntaxNode node, ISyntaxFactsService syntaxFacts)
+        {
             // `var a = b`;
-            // -> `b`
             if (syntaxFacts.IsLocalDeclarationStatement(node))
             {
+                // Check if there's only one variable being declared, otherwise following transformation
+                // would go through which isn't reasonable since we can't say the first one specifically
+                // is wanted.
+                // `var a = 1, `c = 2, d = 3`;
+                // -> `var a = 1`, c = 2, d = 3;
                 var variables = syntaxFacts.GetVariablesOfLocalDeclarationStatement(node);
                 if (variables.Count == 1)
                 {
                     var declaredVariable = variables.First();
-                    var initializer = syntaxFacts.GetInitializerOfVariableDeclarator(declaredVariable);
 
+                    // -> `a = b`
+                    yield return declaredVariable;
+
+                    // -> `b`
+                    var initializer = syntaxFacts.GetInitializerOfVariableDeclarator(declaredVariable);
                     if (initializer != default)
                     {
                         var value = syntaxFacts.GetValueOfEqualsValueClause(initializer);
                         if (value != default)
                         {
-                            return value;
+                            yield return value;
                         }
                     }
                 }
             }
 
+            // var `a = b`;
+            // -> `var a = b`;
+            if (node.Parent != null && syntaxFacts.IsLocalDeclarationStatement(node.Parent))
```

Comment:
`syntaxFacts.IsLocalDeclarationStatement(node?.Parent)`?

Reply:
The helpers don't really work well with nulls. And generally, SyntaxFacts is a bit of a mess in that regard (some helpers handle null, some don't). C# 8 with nullability can't come soon enough.

Reply:
Nvm, it's an extension method down the road so it will actually work with null passed in but it feels weird.

Reply:
FWIW, it's a common and (mostly) consistent pattern in roslyn.  any `IsXXX` function should accept (and almost always return false with) null args.  Basically, the `IsXXX` is often to simulate `x is y`.  In those cases, you can pass in `null` for `x`, and it should return `false` accordingly. :)

Reply:
Also, in terms of `and (mostly) consistent`. If you find an `IsXXX` function that throws on a null input, we really should just fix that method.  I would consider those methods inconsistent with our general designs and patterns, and we should be fixing them rather than updating our calling code to work around it.